### DISCUSSION TITLE
Installation and deployment QA summary

### DISCUSSION THREAD

**Discussion by KingSkyLi** [TuGraph-family/tugraph-db#103] [Q&A]
本贴是安装部署的QA汇总贴，详见下方回答。
This post is a QA summary post for installation and deployment, see the answer below for details.

**Comment by KingSkyLi** [comment]
中文：
Q：如何使用 docker 镜像安装？
A：
   - 确认本地是否有 docker 环境，可使用`docker -v`进行验证。如果没有请安装 docker，安装方式见 docker 官网文档[https://docs.docker.com/install/](https://docs.docker.com/install/) 。
   - 下载 docker 镜像，下载方式可使用`docker pull tugraph/tugraph-runtime-centos7`，也可以在官网下载页面进行下载[https://www.tugraph.org/download](https://www.tugraph.org/download)**[注：下载的文件是*.tar.gz 的压缩包，不用解压]**。
   - 如果使用 docker pull 下载的镜像则不用导入镜像。如果使用官网下载的压缩包，则要使用`docker load -i ./tugraph_x.y.z.tar`**[注：x.y.z 是版本号的代替符，具体数值根据自己下载的版本进行改写]**
   - 启动 docker 容器`docker run -d -p 7070:7070 -p 9090:9090 --name tugraph_demo tugraph/tugraph-runtime-centos7 lgraph_server`**[注：具体的镜像名称 tugraph/tugraph-runtime-centos7 要以本地实际镜像名称为准，可用过 docker images 命令查看]**


English:
Q: How to install using docker image?
A:
   - Confirm whether there is a docker environment locally, you can use `docker -v` to verify. If not, please install docker. For the installation method, see docker official website documentation [https://docs.docker.com/install/](https://docs.docker.com/install/).
   - Download the docker image, you can use `docker pull tugraph/tugraph-runtime-centos7` to download, or you can download it from the official website download page [https://www.tugraph.org/download](https://www.tugraph .org/download)**[Note: The downloaded file is a *.tar.gz compressed package, no need to decompress]**.
   - If you use the image downloaded by docker pull, you don’t need to import the image. If you use the compressed package downloaded from the official website, you need to use `docker load -i ./tugraph_x.y.z.tar`**[Note: x.y.z is a substitute for the version number, and the specific value is rewritten according to the version you downloaded]**
   - Start the docker container `docker run -d -p 7070:7070 -p 9090:9090 --name tugraph_demo tugraph/tugraph-runtime-centos7 lgraph_server`**[Note: The specific image name tugraph/tugraph-runtime-centos7 should start with The actual local image name shall prevail, which can be viewed with the docker images command]**

**Comment by KingSkyLi** [comment]
Q：rpm 包和 deb 包安装后，启动 lgraph_server 服务。提示缺少'liblgraph.so'报错？
A：此问题主要是环境变量导致，需要配置环境量。
   示例：
   ```shell
   export LD_LIBRARY_PATH=/usr/local/lib64
   ```

**Comment by zhangwh807** [comment]
中文：
Q：如何打开审计功能
A：需要打开enable_audit_log开关，具体参考" https://github.com/TuGraph-family/tugraph-db/blob/master/docs/zh-CN/source/10.permission/5.log.md " 中的审计日志

English:
Q: How do I enable the audit feature?
A: You need to turn on the enable_audit_log switch. For details, refer to the Audit Log section at " https://github.com/TuGraph-family/tugraph-db/blob/master/docs/en-US/source/10.permission/5.log.md "

**Comment by zhangwh807** [comment]
中文：
Q：产品是否支持麒麟操作系统？只有企业版支持么？
A：开源和企业版都支持

English:
Q: Does the product support the Kylin operating system? Is it only supported by the enterprise edition?
A: Both the open-source and enterprise editions are supported.

**Comment by zhangwh807** [comment]
中文：
Q：4.3.2版本的镜像，启动成功了，进入容器后没有 setup.sh 脚本是目录换了吗？
A：从TuGraph 4.3.2版本及其后续版本开始，设计上已经简化了启动流程。当你运行TuGraph的Docker镜像之后，图数据库服务和Web浏览器（Browser）会自动启动，无需再手动执行额外的启动脚本如setup.sh。这样的改进旨在为用户提供更加即开即用的体验，减少了部署和初始化的复杂度。

English:
Q: I successfully started the 4.3.2 version image, but there is no setup.sh script when I enter the container. Has the directory changed?
A: Starting from TuGraph version 4.3.2 and onwards, the startup process has been simplified by design. When you run the TuGraph Docker image, the graph database service and the web browser (Browser) will automatically start, so there is no need to manually execute additional startup scripts like setup.sh. This improvement aims to provide users with a more ready-to-use experience, reducing the complexity of deployment and initialization.

**Comment by zhangwh807** [comment]
中文：
Q：启动TuGraph的时候报这个错误：0x00007f7e5f272900 FATAL include/fma-common/binary_buffer.h:289] CHECK(gpos_ + size <= ppos_)  failedreading beyond the array: required size=4, actual size=2
A：因版本升级幅度较大而导致的兼容性问题，建议先清除原有的数据目录（默认是/var/lib/lgraph/data），然后再进行重试。请注意，此操作将永久删除所有存储在该路径下的数据，请确保已做好相应数据备份。

English:
Q: When starting TuGraph, I encountered this error: 0x00007f7e5f272900 FATAL include/fma-common/binary_buffer.h:289] CHECK(gpos_ + size <= ppos_) failed reading beyond the array: required size=4, actual size=2
A: This is a compatibility issue due to a significant version upgrade. It is recommended to clear the existing data directory (default is /var/lib/lgraph/data) and then try again. Please note that this operation will permanently delete all data stored in that path, so ensure you have made the appropriate data backups.

**Comment by zhangwh807** [comment]
中文：
Q：TuGraph版本更新之后，原库的数据会丢吗？
A：按照官方手册启动容器的时候会把数据目录映射到物理机的/root/tugraph/data目录，只要没有删除物理机/root/tugraph/data目录数据就不会丢。

English：
Q: Will data from the original database be lost after updating the TuGraph version?
A: When starting the container according to the official documentation, the data directory will be mapped to the /root/tugraph/data directory on the host machine. As long as you do not delete the data in the /root/tugraph/data directory on the host machine, the data will not be lost.

**Comment by zhangwh807** [comment]
中文：
Q：上传插件，系统提示无权限，是需要哪里配置吗？
A：如果是容器部署的4.5.0版本可以在/usr/local/etc/lgraph.json中加上`"enable_plugin" :true,`这个参数，然后`docker restart tugraph_demo`。
English：
Q: When uploading the plugin, the system prompts that there is no permission. Is there any configuration needed?
A: If you are using the container-deployed version 4.5.0, you can add the parameter `"enable_plugin": true ,`in the file /usr/local/etc/lgraph.json, and then restart the Docker container using `docker restart tugraph_demo`.
